update NodeJs / ExpressJs Tutorial (#26)
HelioNetworks/heliohost-wiki: tutorials/node.js/express.js.md

**Note: delete the content inside `httpdocs`**

if you don't use the example Create a new file `app.js` and put this inside:

```javascript
const http = require('http');
const express = require('express');
const port = 3000;

const app = express();

app.get('/', (req, res) => {
  res.status(200).send('Node.js is working with Express and http.');
});

// Create an HTTP server using the Express app
const server = http.createServer(app);

server.listen(port, () => {
  console.log(`Server running`);
});
```

Create a new file `package.json` and put this inside:

```javascript
{
  "name": "node-test",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "start": "node app.js"
  },
  "dependencies": {
    "cookie-parser": "~1.4.4",
    "debug": "~2.6.9",
    "express": "~4.16.1",
    "http-errors": "~1.6.3",
    "morgan": "~1.9.1",
    "pug": "^3.0.2"
  }
}
```

In Plesk, go to "Website & Domains".

![](../../.gitbook/assets/WebsitesAndDomains.png)

Under the domain you'd like to use for Node.js, select `Node.js`.

![](../../.gitbook/assets/NodejsLink.png)

## Server setup
Upload all your files to your Tommy server. Then go to the Node.js section and set the following.

> * Node.js Version: At least 14.21.0 tho i highly recommend using the latest one 17.9.1

Reviewer: ```suggestion
* Node.js Version: minimum 14.21.0, though it is recommend to use the latest available, currently 17.9.1.
```

> * Application Root: / (folder where you uploaded your next project, i recommend / )

Reviewer: ```suggestion
* Application Root: `/` (This is the folder where you uploaded your project)
```

> * Application Startup File: app.js (file we created earlier)

Reviewer: ```suggestion
* Application Startup File: `app.js` (The entry point of our app, use the file we created earlier)
```

* Here, you're able to configure settings for your application.

![](../../.gitbook/assets/NodejsSettings.png)

## Starting the server

Once all of the above steps are done, you can press the Enable Node.js button.

![](../../.gitbook/assets/enableNodeJs.png)

Then you need to press the NPM install button.

![](../../.gitbook/assets/NPMInstall.png)

After that finished you just need to wait for the server to update your account and start using Node.js, this may take a while.

In order for the Node.js application to be deployed, it requires an Apache restart, so this could take anywhere from a few minutes to a few hours. If it's been more than 2 hours, and it still isn't working, please
open [a customer service ticket](https://www.helionet.org/index/forum/45-customer-service/) and let us know.

> Once the Node.js application has been deployed you should be able to go to `domain.helioho.st/` and see the

Reviewer: ```suggestion
Once the Node.js application has been deployed you should be able to go to the domain you selected and see the
```
